spatial/DMeshAABBTree.cs

```diff
@@ -196,31 +223,41 @@ protected void find_nearest_tri(int iBox, Vector3d p, ref double fNearestSqr, re
         /// <summary>
         /// Find the vertex closest to p, within distance fMaxDist, or return InvalidID
         /// </summary>
-        public virtual int FindNearestVertex(Vector3d p, double fMaxDist = double.MaxValue)
+        public virtual int FindNearestVertex(Vector3d p, double fMaxDist = double.MaxValue,
+            TriangleFilterDelegate triangleFilterF = null,
+            VertexFilterDelegate vertexFilterF = null)
         {
             if (mesh_timestamp != mesh.ShapeTimestamp)
                 throw new Exception("DMeshAABBTree3.FindNearestVertex: mesh has been modified since tree construction");
 
             double fNearestSqr = (fMaxDist < double.MaxValue) ? fMaxDist * fMaxDist : double.MaxValue;
             int vNearID = DMesh3.InvalidID;
-            find_nearest_vtx(root_index, p, ref fNearestSqr, ref vNearID);
+            find_nearest_vtx(root_index, p,
+                triangleFilterF: triangleFilterF ?? (TriangleFilterF != null ? TriangleFilterF.Invoke : (TriangleFilterDelegate)null), // could not simply cast two different delegates
```

Comment:
Can we just remove an odd global `TriangleFilterF` that I had to revert after every filtering operation? By the way, why the delegates are different if, according to the usage, their signature is identical?